### ISSUE TITLE
Black overlay covering whole screen if VLC is not installed

### ISSUE THREAD

**Issue by GauravButola** [butterproject/butter-desktop-angular#23]
If VLC is not already installed, then `ptDetail` directive crashes, due to which screen is covered with a black overlay making the app unusable from the start.

An error pops up in the console:
`libvlc.so.5: cannot open shared object file: No such file or directory`

This issue goes away upon installing VLC which effectively makes `liblvlc.so` available.

**Comment by hadim** [comment]
Well ptDetail is related to webchimera and for now this module is broken and not working. In the future it is planned for butter to use this https://github.com/BoLaMN/wcjs-angular (which is largely based on the webchimera inside this repo).

If you want to hack on the player, please talk to @BoLaMN.

**Comment by BoLaMN** [comment]
OS X or nix @GauravButola ?

**Comment by GauravButola** [comment]
@BoLaMN Ubuntu 15.04.

**Comment by sinas06** [comment]
Same issue overhere, but i have VLC installed. Using Windows 10.

**Comment by BoLaMN** [comment]
My only thoughts are the wcjs-prebuilt module isn't detecting correctly, might be time to switch over to my grunt-wcjs version I made here https://github.com/bolamn/grunt-wcjs example grunt config here https://github.com/BoLaMN/wcjs-angular-example/blob/master/Gruntfile.coffee would take extra work to modify electron-livereload as well to handle the location of the deps

**Comment by hadim** [comment]
I also notive the overlay with packaged applications (using `grunt package`).

**Comment by conductingchaos** [comment]
this is what I'm seeing upon executing 'grunt start' from Git CMD or Git Bash. I just updated to the most recent version of VLC. Windows 8.1 x64. How do I access the console or log to provide better feedback?

![butterbuilt](https://cloud.githubusercontent.com/assets/15332100/10743079/3b83e85c-7c07-11e5-8fc2-a53823d5a791.jpg)

**Comment by conductingchaos** [comment]
never mind I opened ' ./scripts/menus/win32.coffee ' with Visual Studio and found the shortcut for enabling dev tools. Here's what I'm getting for errors: 
![butterdevlog](https://cloud.githubusercontent.com/assets/15332100/10743940/b0803f16-7c0c-11e5-830d-790f92eebd84.jpg)

I have a feeling that you're already aware of this, I'm going to try messing with it though. I haven't coded anything in Java since like 2006 and I've never written applications in Javascript. I saw there might be some python in here somewhere as well? So I'm not expecting to really be able to help much but I'm sick and bored so might as well do something.

**Comment by hadim** [comment]
It's not python but Coffee. Coffee is a language that is converted to javascript. So basically is JS but with a python syntax :-)

Anyway I will try to reproduce the error.

**Comment by hadim** [comment]
What version of node are you using (`node --version`) ?

**Comment by conductingchaos** [comment]
I installed node 4.2.1

**Comment by hadim** [comment]
@BoLaMN I don't understand why webchimera.html is included in the detail page of a movie since the movie can be played with an external player... so the internal player should be loaded later isnt it ?

**Comment by hadim** [comment]
@GauravButola since https://github.com/butterproject/butter-desktop-angular/commit/3aea7096782fb2df92dbbc0f6ff84fd76c9f2075, it should work even if vlc is not installed. Could you try please ?

**Comment by SamYStudiO** [comment]
hi, got same pb here with win10

**Comment by conductingchaos** [comment]
WebChimera, Grunt, Electron, and Angular are all supporting packages that are prebuilt and we don't need to touch to make this work? 

WebChimera Angular was written by BoLaMN, a PCT / butter developer. is that the files I should then be looking at?

**Comment by hadim** [comment]
I don't know guys but for now angular repo is very ealry stage so maybe you should come back in a while and see if there is improvments on this.

**Comment by conductingchaos** [comment]
ok but this will be the new main fork? I can't clone the non angular butter folder from git due to remaining references to PCT.io servers.

**Comment by hadim** [comment]
The main repo is butter and you should try. It'll work better than this one
Le 27 oct. 2015 12:12 AM, "conductingchaos" notifications@github.com a
écrit :

**Comment by bartlangelaan** [comment]
I have the same issue.

Windows 10 Pro, 64 bit
VLC v2.2.1
Node v4.0.0
Npm v2.14.2

Install log: http://pastebin.com/yhdE4XtG
Error log: http://pastebin.com/NCMRi4R9

If you need any more info, please ask.

**Comment by GauravButola** [comment]
@hadim Didn't seem to be working for me; I have the latest code from master. Attaching the console output screenshot if that helps.

Maybe for me it's not working because I was fiddling around with a lot of stuff, can someone else confirm if it works for them?

![screenshot from 2015-10-27 11 29 23](https://cloud.githubusercontent.com/assets/734636/10750647/52a1d106-7c9e-11e5-9135-30994d6b0ce5.png)

**Comment by paularchive** [comment]
It has nothing to do with rather that vlc is installed or not..i assume you have a 64-bit windows installation. And if you look at the wcjs-prebuilt repo on github u will find out that this is a bug for 64-bit windows users. @BoLaMN is already looking into this

**Comment by hadim** [comment]
Nice catch @paulhobbel . Issue seems to come from here https://github.com/Ivshti/wcjs-prebuilt/issues/10

You guys can try idea posted on the below issue and see wether it's work for you or not.

**Comment by hadim** [comment]
@GauravButola I think you are facing an other issue. Are you running linux ?

**Comment by GauravButola** [comment]
Yes @hadim,  I'm running Linux, Ubuntu 15.04 to be specific. I don't face the issue after installing VLC though.

**Comment by hadim** [comment]
@GauravButola I don't understand because your logs are saying that vlc is detected...

**Comment by GauravButola** [comment]
@hadim Seems like I've broken VLC on my system, it won't open at all now. So, now there's something wrong in part as well. Initially it was not detecting VLC at all (because it was not there), so the logs above about VLC were not there for me when I reported the bug.

I guess if someone else isn't facing the same issue, we can close the issue.

**Comment by hadim** [comment]
Well your system installed vlc shouldn't interfer with Butter and should not produce this bug I guess... But we are far from this kind of details for now.

Let it open for the other issue.

**Comment by bartlangelaan** [comment]
I can confirm that on Windows 10, 64 bit this quickfix is working! Thank you!

Steps to quickfix:
1. Download the 64bit version of VLC: http://get.videolan.org/vlc/2.2.1/win64/vlc-2.2.1-win64.zip
2. In `butter-desktop-angular\node_modules\wcjs-prebuilt\bin` replace `libvlc.ddl`, `libvlccore.ddl` and the `plugins` folder with the 64 bit version of VLC.
